Source/ChocolateyGui/Services/ChocoSettingsSourceService.cs

```diff
@@ -0,0 +1,87 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright company="Chocolatey" file="ChocoSettingsSourceService.cs">
+//   Copyright 2014 - Present Rob Reynolds, the maintainers of Chocolatey, and RealDimensions Software, LLC
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace ChocolateyGui.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Linq;
+    using System.Runtime.InteropServices;
+    using System.Text;
+    using System.Threading.Tasks;
+    using ChocolateyGui.AsyncProcess;
+    using ChocolateyGui.Models;
+    using ChocolateyGui.Providers;
+    using ChocolateyGui.ViewModels.Items;
+
+    public class ChocoSettingsSourceService : ISourceService
+    {
+        private readonly IChocolateyConfigurationProvider chocolateyConfigurationProvider;
+        private string chocoExePath = string.Empty;
+
+        public ChocoSettingsSourceService(IChocolateyConfigurationProvider chocolateyConfigurationProvider)
+        {
+            this.chocolateyConfigurationProvider = chocolateyConfigurationProvider;
+            this.chocoExePath = Path.Combine(this.chocolateyConfigurationProvider.ChocolateyInstall, "choco.exe");
+        }
+
+        public event SourcesChangedEventHandler SourcesChanged;
+
+        public async void AddSource(SourceViewModel sourceViewModel)
+        {
+            if (sourceViewModel == null)
+            {
+                throw new ArgumentNullException("sourceViewModel");
+            }
+
+            var arguments = new StringBuilder();
+            arguments.AppendFormat("source add {0} {1} -y", sourceViewModel.Name, sourceViewModel.Url);
+
+            await ProcessEx.RunAsync(this.chocoExePath, arguments.ToString());
+
+            var sourcesChangedEvent = this.SourcesChanged;
+            if (sourcesChangedEvent != null)
+            {
+                sourcesChangedEvent(
+                    this,
+                    new SourcesChangedEventArgs(new List<SourceViewModel> { sourceViewModel }, new List<SourceViewModel>()));
+            }
+        }
+
+        public SourceViewModel GetDefaultSource()
+        {
+            throw new System.NotImplementedException();

```

Comment:
perhaps putting a message in the ctor of the exception? or a // TODO above? Just so we don't forget :)